pywbem_mock/_instancewriteprovider.py

```diff
@@ -270,367 +209,280 @@ def __init__(self, cimrepository=None):
         Parameters:
 
           cimrepository (:class:`~pywbem_mock.BaseRepository` or subclass):
-            Defines the repository to be used by request responders.  The
-            repository is fully initialized.
+            The CIM repository to be used by mock WBEM server responders.
+            The repository must be fully initialized.
         """
         super(InstanceWriteProvider, self).__init__(cimrepository)
 
     ####################################################################
     #
-    #   CreateInstance server request
+    #   CreateInstance mock WBEM server responder
     #
     ####################################################################
 
-    def CreateInstance(self, namespace, NewInstance):
+    def CreateInstance(self, namespace, new_instance):
         """
-        Implements a WBEM server responder for
+        Implements a mock WBEM server responder for
         :meth:`pywbem.WBEMConnection.CreateInstance`.
 
-        Create a CIM instance in the local CIM repository of this class.
-
-        The input parameters must have been validated before this method
-        is called including:
-
-        1. Namespace is valid in CIM repository.
-        2. The ``NewInstance`` is a valid CIM Instance.
-        3. All key properties defined in the class  of ``NewInstance`` are
-           in ``NewInstance``. This method cannot add properties or
-           property values to ``NewInstance``.
-        4. There are no properties in ``NewInstance`` that are not in the
-           corresponding class.
-        5. NewInstance path must not exist in the CIM repository.
+        Create a new CIM instance in the CIM repository of the mock WBEM server.
+
+        Validation already performed by the provider dispatcher that calls
+        this provider method:
+        - The provider method is called only for the registered class and
```

Comment:
I think this applies to user-defined providers only.

Reply:
DONE - good catch. Corrected.